source/common/version/version.cc

```diff
@@ -37,10 +37,11 @@ const envoy::config::core::v3::BuildVersion& VersionInfo::buildVersion() {
 }
 
 bool VersionInfo::sslFipsCompliant() {
+  const std::string fips_ssl_version = "BoringSSL-FIPS";
   bool fipsCompliant = false;
-#ifdef BORINGSSL_FIPS
-  fipsCompliant = true;
-#endif
+  if (VersionInfo::sslVersion() == fips_ssl_version) {
```

Comment:
This reverts back to the string matching that you originally had in #14719, which @ggreenway and @lizan recommended against.

Note that `#ifdef BORINGSSL_FIPS` doesn't work here, since `//source/common/version` doesn't depend on BoringSSL library, since that wouldn't work for OpenSSL for version string.

I think you should add a new `ENVOY_SSL_IS_FIPS` `define` and use it here.

Reply:
@PiotrSikora Thanks for the suggestion. I will try this approach.